src/server/req_manager.c

```diff
@@ -2449,10 +2673,169 @@ make_host_addresses_list(char *phost, u_long **pul)
 		(*pul)[i] = 0; /* null term array ip adrs */
 
 		freeaddrinfo( pai);
-		return 0;
 #endif
+		tpul = malloc(sizeof(struct pul_store));
+		if (!tpul) {
+			strcat(log_buffer, "out of  memory");
+			return (PBSE_SYSTEM);
+		}
+		tpul->len = len;
+		tpul->pul = (u_long *) malloc(tpul->len);
+		if (!tpul->pul) {
+			free(tpul);
+			strcat(log_buffer, "out of  memory");
+			return (PBSE_SYSTEM);
+		}
+		memmove(tpul->pul, *pul, tpul->len);
+
+		if (hostaddr_tree == NULL ) {
+			hostaddr_tree = cr_tree(0);
+			if (hostaddr_tree == NULL ) {
+				free(tpul->pul);
+				free(tpul);
+				strcat(log_buffer, "out of  memory");
+				return (PBSE_SYSTEM);
+			}
+		}
+		if (tree_add_del(hostaddr_tree, phost, tpul, 0) != 0) {
+			free(tpul->pul);
+			free(tpul);
+			return (PBSE_SYSTEM);
+		}
+		return 0;
+}
+
+/**
+ * @brief Create a new AVL tree
+ *
+ * @param[in] dups - Are duplicate entries allowed
+ *
+ * @return         Ptr to the root of the tree
+ * @retval !NULL - The pointer to the root of the tree
+ * @retval NULL  - Failure
+ *
+ * @par MT-safe: No
+ **/
+AVL_IX_DESC *
+cr_tree(int dups)
```

Comment:
avltree.c covers the index routines, and was written by Gregory Tseytin.  I am hesitant to add to it...should I?